src/main.rs

```diff
@@ -185,15 +212,17 @@ fn run_test_case(results: &mut Results,
                  extra_client_args: &Vec<String>,
                  extra_server_args: &Vec<String>) {
 
-    let r = run_test_case_inner(config, case, extra_client_args, extra_server_args);
-    results.update(case, index, &r);
+    match run_test_case_inner(config, case, extra_client_args, extra_server_args) {
+        Ok((c, s)) => results.update(case, index, &c, &s),
+        Err(e) => results.update_simple(case, index, &TestResult::from_status(e))
```

Comment:
I don't think I understand why we need `update_simple` in addition to `update`.

Reply:
Merged update() and update_simple() into a single function again that can handle both types of results. The whole logic is still not very concise though, I might improve that later in an attempt to make constructing and running test cases a little more rusty.